_delb/xpath.py

```diff
@@ -74,6 +79,10 @@ def _partition_terms(expression: str) -> List[str]:  # noqa: C901
             elif character == "(":
                 bracket_level += 1
             elif character == ")":
+                if bracket_level == function_level and any(
+                    current_term.startswith(func) for func in BOOLEAN_FUNCTIONS
```

Comment:
neat, that's much more clever than my idea of looking ahead.

Reply:
thanks, obviously i considered lookahead at first as well, before realizing that with everything you already had put in place we could just look behind and be done with it `:3`

Reply:
i'm not sure as to what consequences might arise from me having forgotten to reset `function_level` after exiting a function expression...